components/movementsensor/gpsrtkpmtk/gpsrtkpmtk_test.go

```diff
@@ -0,0 +1,328 @@
+package gpsrtkpmtk
+
+import (
+	"context"
+	"math"
+	"testing"
+
+	"github.com/edaniels/golog"
+	geo "github.com/kellydunn/golang-geo"
+	"go.viam.com/test"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/components/movementsensor/fake"
+	gpsnmea "go.viam.com/rdk/components/movementsensor/gpsnmea"
+	rtk "go.viam.com/rdk/components/movementsensor/rtkutils"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/testutils/inject"
+)
+
+const (
+	testRoverName   = "testRover"
+	testStationName = "testStation"
+	testBoardName   = "board1"
+	testBusName     = "bus1"
+	testi2cAddr     = 44
+)
+
+func TestValidateRTK(t *testing.T) {
+	path := "path"
+	cfg := Config{
+		NtripURL:             "http//fakeurl",
+		NtripConnectAttempts: 10,
+		NtripPass:            "somepass",
+		NtripUser:            "someuser",
+		NtripMountpoint:      "NYC",
+		Board:                testBoardName,
+		I2CBus:               testBusName,
+		I2CAddr:              testi2cAddr,
+	}
+	t.Run("valid config", func(t *testing.T) {
+		err := cfg.validateNtrip(path)
+		test.That(t, err, test.ShouldBeNil)
+		err = cfg.validateI2C(path)
+		test.That(t, err, test.ShouldBeNil)
+		_, err = cfg.Validate(path)
+		test.That(t, err, test.ShouldBeNil)
+	})
+
+	t.Run("invalid ntrip url", func(t *testing.T) {
+		cfg := Config{
+			NtripURL:             "",
+			NtripConnectAttempts: 10,
+			NtripPass:            "somepass",
+			NtripUser:            "someuser",
+			NtripMountpoint:      "NYC",
+			Board:                testBoardName,
+			I2CBus:               testBusName,
+			I2CAddr:              testi2cAddr,
+		}
+		err := cfg.validateNtrip(path)
```

Comment:
is it necessary to call the Ntrip and i2c validations each time? I think they will be called when you call the `Validate` function